app/controllers/application_controller.rb

```diff
@@ -26,6 +27,17 @@ class ApplicationController < ActionController::Base
                 :theme_stylesheet_url,
                 :extension_javascript_url
 
+  def should_redirect?
+    should_choose_organization? && current_user.accessible_organizations.length == 1
```

Comment:
Please delegate to `current_user.has_main_organization?`

Reply:
That way you can remove logic from controller, move it to model and test it easily.